### DISCUSSION TITLE
Variants as boolean

### DISCUSSION THREAD

**Discussion by lobor** [jpudysz/react-native-unistyles#118] [Q&A]
Hi, how are you ?

I'm come back for another question 😀 (related to https://github.com/jpudysz/react-native-unistyles/discussions/115)

I try to set a variant as boolean

Base code
```tsx
const example = ({ brand = false }: { brand?: boolean } = {}) => {
  useStyles(sheet, { brand });
};

const sheet = createStyleSheet({
  base: {
    variants: {
      brand: {},
    },
  },
});
```
with this code, i have typescript error (normal behavior)
![image](https://github.com/jpudysz/react-native-unistyles/assets/3316940/cc7bdea3-eb95-4d6a-a00e-57009ddd8586)

When i put `true` typescript detect as string
![image](https://github.com/jpudysz/react-native-unistyles/assets/3316940/8cf5cda6-0886-4f97-a930-89d1da5fabbb)

Is there currently a way to do this?

I suppose should change logic parsing and typescript type

**Comment by jpudysz** [comment]
Hello 👋,

This is an incorrect method of handling variants. Variants should belong to variant groups (such as “brand”), and then have variants (like “true”) with valid RN styles.

To select the variant you named “true,” you need to pass the string “true.” You cannot cast it to a Boolean.

Variants cannot be selected using booleans. They must be selected by their names (strings).

**Comment by lobor** [comment]
In the future, will this be a feature that will be implemented?

**Reply by jpudysz** [reply]
No, I don’t think so. You can create multiple variants and select them with a string. For a Boolean, you would need to have only two variants, which is an unnecessary restriction.

**Comment by jpudysz** [comment]
I received another request for it, I'm willing to support it soon. Thanks for your feedback!

**Comment by jpudysz** [comment]
PR: https://github.com/jpudysz/react-native-unistyles/pull/134
